azure-mgmt-resources/src/main/java/com/azure/management/resources/fluentcore/utils/ProviderRegistrationPolicy.java

```diff
@@ -0,0 +1,103 @@
+            response -> {
+                if (!responseSuccessful(response)) {
+                    HttpResponse bufferedResponse = response.buffer();
+                    return FluxUtil.collectBytesInByteBufferStream(bufferedResponse.getBody()).flatMap(
+                        body -> {
+                            String bodyStr = new String(body, StandardCharsets.UTF_8);
+
+                            AzureJacksonAdapter jacksonAdapter = new AzureJacksonAdapter();
+                            CloudError cloudError;
+                            try {
+                                cloudError = jacksonAdapter.deserialize(bodyStr, CloudError.class, SerializerEncoding.JSON);
+                            } catch (IOException e) {
+                                return Mono.just(bufferedResponse);
+                            }
+
+                            if (cloudError != null && MISSING_SUBSCRIPTION_REGISTRATION.equals(cloudError.getCode())) {
+                                String subscriptionId = ResourceUtils.extractFromResourceId(request.getUrl().getPath(), "subscriptions");
+                                RestClient restClient = new RestClientBuilder()
+                                        .withBaseUrl(String.format("%s://%s", request.getUrl().getProtocol(), request.getUrl().getHost()))
+                                        .withCredential(credential)
+                                        .withSerializerAdapter(jacksonAdapter).buildClient();
+                                ResourceManager resourceManager = ResourceManager.authenticate(restClient)
+                                        .withSubscription(subscriptionId);
+                                Pattern providerPattern = Pattern.compile(".*'(.*)'");
+                                Matcher providerMatcher = providerPattern.matcher(cloudError.getMessage());
+                                providerMatcher.find();
+                                Provider provider = registerProvider(providerMatcher.group(1), resourceManager);
+                                while (provider.registrationState().equalsIgnoreCase("Unregistered")
+                                    || provider.registrationState().equalsIgnoreCase("Registering")) {
+                                    SdkContext.sleep(5 * 1000);
+                                    provider = resourceManager.providers().getByName(provider.namespace());
+                                }
+
+                                // Retry after registration
+                                return next.process();
+                            }
+                            return Mono.just(bufferedResponse);
+                        }
+                    );
+                }
+                return Mono.just(response);
+            }
+        );
+    }
+
+    private Provider registerProvider(String namespace, ResourceManager resourceManager) {
+        return resourceManager.providers().register(namespace);
```

Comment:
You seems calling a sync version? Should use the async?

Reply:
Done